mne/preprocessing/nirs/_temporal_derivative_distribution_repair.py

```diff
@@ -0,0 +1,143 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Frank Fishburn
+#
+# License: BSD (3-clause)
+
+
+import numpy as np
+
+from ... import pick_types
+from ...io import BaseRaw
+from ...utils import _validate_type
+
+
+def temporal_derivative_distribution_repair(raw):
+    r"""Apply temporal derivative distribution repair to data.
+
+    Applies temporal derivative distribution repair (TDDR) to data.
+    :footcite:`FishburnEtAl2019`. This approach removes baseline shift
+    and spike artifacts without the need for any user-supplied parameters.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of Raw
+         Data with TDDR applied.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    raw = raw.copy().load_data()
+    _validate_type(raw, BaseRaw, 'raw')
+
+    if not len(pick_types(raw.info, fnirs='fnirs_od')):
+        raise RuntimeError('TDDR should be run on optical density data.')
+
+    raw._data = _TDDR(raw._data.T, raw.info['sfreq']).T
+
+    return raw
+
+
+# Taken from https://github.com/frankfishburn/TDDR/ (MIT license).
+# With permission https://github.com/frankfishburn/TDDR/issues/1.
+# The only change to the code is the function name and the recursive call
+# in the fifth line of code.
+def _TDDR(signal, sample_rate):
+    # This function is the reference implementation for the TDDR algorithm for
+    #   motion correction of fNIRS data, as described in:
+    #
+    #   Fishburn F.A., Ludlum R.S., Vaidya C.J., & Medvedev A.V. (2019).
+    #   Temporal Derivative Distribution Repair (TDDR): A motion correction
+    #   method for fNIRS. NeuroImage, 184, 171-179.
+    #   https://doi.org/10.1016/j.neuroimage.2018.09.025
+    #
+    # Usage:
+    #   signals_corrected = TDDR( signals , sample_rate );
+    #
+    # Inputs:
+    #   signals: A [sample x channel] matrix of uncorrected optical
+    #            density data
+    #   sample_rate: A scalar reflecting the rate of acquisition in Hz
+    #
+    # Outputs:
+    #   signals_corrected: A [sample x channel] matrix of corrected
+    #   optical density data
+
+    from scipy.signal import butter, filtfilt
+    signal = np.array(signal)
+    if len(signal.shape) != 1:
+        for ch in range(signal.shape[1]):
+            signal[:, ch] = _TDDR(signal[:, ch], sample_rate)
+        return signal
```

Comment:
If this is how it operates, it would be simpler not to make this modification, and do the looping over `picks` directly in `temporal_derivative_distribution_repair`. It will hopefully keep this function just a copy-paste drop-in (?) and also takes care of the `picks` problem at the same time.

Reply:
This still required the name change to keep flake happy unfortunately.